src/main/java/network/brightspots/rcv/GuiTabulateController.java

```diff
@@ -0,0 +1,410 @@
+   *
+   * @param keyEvent ignored
+   */
+  public void nameUpdated(KeyEvent keyEvent) {
+    updateGuiWithNameEnteredStatus();
+    setTabulationButtonStatus();
+    updateProgressText();
+  }
+
+  /**
+   * Action when the Load CVRs button is clicked.
+   *
+   * @param actionEvent ignored
+   */
+  public void buttonLoadCvrsClicked(ActionEvent actionEvent) {
+    String configPath = getConfigPathOrCreateTempFile();
+
+    enableButtonsUpTo(null);
+    tabulateButton.setText("Tabulate");
+    Service<LoadedCvrData> service = guiConfigController.parseAndCountCastVoteRecords(configPath);
+
+    watchParseCvrServiceProgress(service);
+  }
+
+  /**
+   * Action when the tabulate button is clicked.
+   *
+   * @param actionEvent ignored
+   */
+  public void buttonTabulateClicked(ActionEvent actionEvent) {
+    String configPath = getConfigPathOrCreateTempFile();
+
+    Service<Boolean> service =
+        guiConfigController.startTabulation(
+            configPath,
+            userNameField.getText(),
+            useTemporaryConfigBeforeTabulation,
+            lastLoadedCvrData);
+
+    watchTabulatorServiceProgress(service);
+  }
+
+  /**
+   * Action when Open Results button is clicked.
+   *
+   * @param actionEvent ignored
+   */
+  public void buttonOpenResultsClicked(ActionEvent actionEvent) {
+    if (lastTaskSucceeded) {
+      openOutputDirectoryInFileExplorer();
+    } else {
+      // Close the window
+      Window window = tabulateButton.getScene().getWindow();
+      ((Stage) window).close();
+    }
+  }
+
+  private void watchTabulatorServiceProgress(Service<Boolean> service) {
+    // Measure time it takes for the function to complete
+    long startTime = System.currentTimeMillis();
+
+    // Prevent the user from closing the window while tabulation is in progress
+    Window window = tabulateButton.getScene().getWindow();
+    Stage stage = ((Stage) window);
+    stage.setOnCloseRequest(event -> event.consume());
+
+    EventHandler<WorkerStateEvent> onSucceededEvent =
+        workerStateEvent -> {
+          lastTaskSucceeded = service.getValue();
+          if (lastTaskSucceeded) {
+            openResultsButton.setText("Open Results Folder");
+          } else {
+            openResultsButton.setText("Close and View Logs");
+          }
+          enableButtonsUpTo(openResultsButton);
+          stage.setOnCloseRequest(null);
+          long endTime = System.currentTimeMillis();
+          Logger.info("Tabulation took " + (endTime - startTime) / 1000.0 + " seconds.");
+        };
+    watchGenericService(service, onSucceededEvent);
+  }
+
+  private void watchParseCvrServiceProgress(Service<LoadedCvrData> service) {
+    EventHandler<WorkerStateEvent> onSucceededEvent =
+        workerStateEvent -> {
+          enableButtonsUpTo(tabulateButton);
+          LoadedCvrData data = service.getValue();
+          tabulateButton.setText("Tabulate " + String.format("%,d", data.numCvrs()) + " ballots");
```

Comment:
I don't know how I feel about this change. What was the motivation for doing it?

1) If someone hot swaps / modifies the file in the background to add or delete ballots (like you described in the other thread), it seems like this is reinforcing an inaccurate perception.
2) If we have a contest with votes in the millions, it truncates the label.

If you want to keep it this way rather than reverting to what we had, I'd recommend at least making it wide enough to accommodate 1 billion votes without truncating.

Reply:
I had a slack conversation (which I will put in the PR review moving forward now that I know exactly how to do that) about this. I thought that when the Tabulate button is enabled and gets colored in following completion of the first step that it drew my attention so much that I didn't notice that the ballot count had been updated in the data above. We decided to put it in this button so it was unmistakeable.

And supporting X,XXX,XXX,XXX means we've futureproofed RCTab for worldwide RCV contests!!!

Reply:
1. If someone hotswaps, it will throw a halting error, and tell them they must reload.
2. I have enabled wordwrap and made the button taller to accommodate trillions of ballots

To piggy-back of what yezr said: previously, the UI tried to get your attention in two places, the Tabulate button and the CVR count. We're really trying to encourage double-checking the counts before hitting Tabulate, but the attention-splitting meant users might not see the updated info. By moving it here, there's a single location the user should look at after hitting tabulate. (Well, in another PR, the table too -- but this value is the key info).